File: 1528-shuffle-string/1528-shuffle-string.java
class Solution {
    public String restoreString(String s, int[] indices) {
        char[] sArray = s.toCharArray();
        char[] restored = new char[sArray.length];
        
        for(int i = 0;i < sArray.length; i++){
            Character character = sArray[i];
            int currentIndex = indices[i];
            restored[currentIndex] = character;
        }
        
        return String.valueOf(restored);
    }
}

File: 1528-shuffle-string/1528-shuffle-string.kt
class Solution {
    fun restoreString(s: String, indices: IntArray): String {
        
        val restored = CharArray(s.length)
        
        s.forEachIndexed{ index, currentChar ->
            val correctIndex = indices[index]
            restored[correctIndex] = currentChar
        }
        
        return String(restored)
    }
}